### PR TITLE
refactor(x/upgrade): voting power threshold 

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2878

### CODE REVIEW

File: x/upgrade/keeper.go [hunk 1]
```diff
@@ -169,15 +175,25 @@ func (k Keeper) TallyVotingPower(ctx sdk.Context, threshold int64) (bool, uint64
...
+	}
+	mod, err := product.SafeMod(sdkmath.NewInt(thresholdFraction.Denominator))
+	if err != nil {
+		return sdkmath.ZeroInt(), err
+	}
+	if mod.IsZero() {
+		return res, nil
 	}
-	return product.QuoRaw(thresholdFraction.Denominator).AddRaw(1)
+	return res.AddRaw(1), nil
```
Comment (rootulp) [inline x/upgrade/keeper.go:196]: This code seems complicated so I'm open to refactoring it if we're okay with dropping the `Fraction` type

Comment (rootulp) [reply x/upgrade/keeper.go]: Found a way to refactor using Cosmos SDK decimal. See 3fdc1d7838338ce74f047c3347619b656b70cf59. Note this results in a small amount of precision loss when dealing with extremely large total voting powers but IMO the precision loss is tolerable in favor of cleaner code. Thoughts @cmwaters ?

File: x/upgrade/keeper.go [hunk 2]
```diff
@@ -16,21 +16,17 @@ var (
 	_ types.MsgServer   = &Keeper{}
 	_ types.QueryServer = Keeper{}
 
-	defaultSignalTheshold = Fraction{Numerator: 5, Denominator: 6}
+	// defaultSignalThreshold is 5/6 or approximately 83.33%
+	defaultSignalThreshold = sdk.NewDec(5).Quo(sdk.NewDec(6))
```
Comment (cmwaters) [inline x/upgrade/keeper.go:20]: Dec has 6 decimals of precision right?

**Review by cmwaters — APPROVED** [review]
Given that we use 5/6ths as the hard coded threshold, I think it is tolerable that there may be some loss in the precision. I don't think it's critical to security that the threshold is exactly 5/6 so to make this cleaner I would agree with this

File: x/upgrade/keeper.go (hunk 2 above)
Comment (rootulp) [reply x/upgrade/keeper.go:20]: Looks like it has more than 6:

```
defaultSignalThreshold (string): 0.833333333333333333
defaultSignalThreshold (float64): 0.8333333333333334
```

**Pull request merged**